### DISCUSSION TITLE
How to create a custom scalar type coercer that renders as JavaScript BigInt?

### DISCUSSION THREAD

**Discussion by learnerEvermore** [graphql-java/graphql-java#2646] [Q&A]
JavaScript [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt)s are one way of solving the `Long` issue, albeit in a way not "exactly" compatible with standard JSON or other languages. JavaScript suffixes those big integer literals with `n` similarly to how Java requires `long` literals to be suffixed by `L`.

I don't know how to write a coercer, however, that would cause `field: 123n` to be included in the output as opposed to `field: 123` or `field: "123"`, neither one of which would be a valid BigInt.

How can this be controlled?

**Comment by bbakerman** [comment]
> JavaScript BigInts are one way of solving the Long issue

Sorry but you havent really outlined your problem - so we are guessing at it.

Do you mean - you want a Scalar type that can handle values in the JS BigInt range?  Something else?  Some other format??

Or you want a scalar that can output integer values in `12345n` format?

I am guessing here obviously.

I will love you can example of custom scalars - in fact a library full of them

https://github.com/graphql-java/graphql-java-extended-scalars/blob/master/src/main/java/graphql/scalars/ExtendedScalars.java#L216
